pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -909,9 +914,17 @@ func (s *DiscoveryServer) startPush(version string, push *model.PushContext, ful
 					pushTimeoutFailures.Add(1)
 					return
 				}
+				if !timer.Stop() {
```

Comment:
We only get here if the timer is stopped so why stop again? and then recieve on the same channel?

Reply:
oh I agree, I was just trying to get one of these code paths to execute in the test case - neither did, so I thought perhaps the goroutine was taking a different case path.